module/VuFindTheme/src/VuFindTheme/ThemeInfo.php

```diff
@@ -210,35 +210,80 @@ public function getThemeInfo()
         return $this->allThemeInfo;
     }
 
+    /**
+     * Determine if a variable is a string-keyed array
+     *
+     * @param mixed $op Variable to test
+     *
+     * @return boolean
+     */
+    protected function isStringKeyedArray($op) {
+        if (!is_array($op)) {
+            return false;
+        }
+
+        reset($op);
+        if (is_string(key($op))) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Add parent theme information to a collection of merged theme info.
+     * Using the string-keyed array format of theme config info,
+     * recursively walk the array, capturing unique or missing values.
+     *
+     * @param array $merged Merged theme info, overrides parent theme
+     * @param array $current Theme info to be merged
+     *
+     * @return array merged theme info, favoring child themes (merged)
```

Comment:
Are these types really correct? Isn't it possible to be array or string in some situations? Are there some edge cases that might be problematic, like if a parent theme contains a string and a child theme contains an array (which might lead to `foreach` being called on a string)?

Sorry if I'm misreading the context in which this is used, but it seems like we might want to tighten this up a little more (and maybe create a weird test case to exercise the potentially problematic edges).